pypeit/specobjs.py

```diff
@@ -484,26 +471,42 @@ def remove_sobj(self, index):
 
 
     def copy(self):
+        """
+        Generate a copy of self
+
+        Returns:
+            SpecObjs
+
+        """
         sobj_copy = SpecObjs()
         for sobj in self.specobjs:
             sobj_copy.add_sobj(sobj.copy())
         sobj_copy.build_summary()
         return sobj_copy
 
     def set_idx(self):
+        """
+        Set the idx in all the SpecObj
+        Update the summary Table
+
+        Returns:
+
+        """
         for sobj in self.specobjs:
             sobj.set_idx()
         self.build_summary()
 
 
     def __getitem__(self, item):
-        """ Overload to allow one to pull an attribute
+        """
```

Comment:
Functions with two leading underscores are basically never documented in sphinx.  Functions with one underscore can be if we request it.  We should put some does at the beginning of the module or in the `__init__` docstring the describes class usage and gives some examples.

Reply:
pushed to `__init__` doc string